app/domain/authentication/authn_k8s/execute_command_in_container.rb

```diff
@@ -0,0 +1,141 @@
+# frozen_string_literal: true
+
+require 'command_class'
+require 'uri'
+require 'websocket'
+require 'rubygems/package'
+
+require 'active_support/time'
+require 'websocket-client-simple'
+
+module Authentication
+  module AuthnK8s
+
+    ExecuteCommandInContainer ||= CommandClass.new(
+      dependencies: {
+        env:                           ENV,
+        websocket_client:              WebSocket::Client::Simple,
+        ws_client_event_handler_class: WebSocketClientEventHandler,
+        message_log_class:             MessageLog,
+        validate_message:              MessageLog::ValidateMessage.new,
+        logger:                        Rails.logger
+      },
+      inputs:       %i(k8s_object_lookup pod_namespace pod_name container cmds body stdin)
+    ) do
+
+      extend Forwardable
+      def_delegators :@k8s_object_lookup, :kube_client
+
+      DEFAULT_KUBE_EXEC_COMMAND_TIMEOUT = 5
+
+      def call
+        init_ws_client
+        add_websocket_event_handlers
+        wait_for_close_message
+        verify_channel_is_closed
+        verify_error_stream_is_empty
+        websocket_messages
+      end
+
+      private
+
+      def init_ws_client
+        ws_client
+        ws_client_event_handler
+      end
+
+      def ws_client
+        @ws_client ||= @websocket_client.connect(server_url, headers: headers)
+      end
+
+      def ws_client_event_handler
+        @ws_client_event_handler ||= @ws_client_event_handler_class.new(
+          ws_client:        ws_client,
+          stdin:            @stdin,
+          body:             @body,
+          pod_name:         @pod_name,
+          validate_message: @validate_message,
+          message_log:      @message_log_class.new,
+          logger:           @logger
+        )
+      end
+
+      def add_websocket_event_handlers
+        # We need to set this as if we'll use @ws_client_event_handler inside
+        # the curly brackets it will look for such a member in the websocket_client
+        # class
+        ws_client_event_handler = @ws_client_event_handler
+
+        ws_client.on(:open) { ws_client_event_handler.on_open }
+        ws_client.on(:message) { |msg| ws_client_event_handler.on_message(msg) }
+        ws_client.on(:close) { ws_client_event_handler.on_close }
+        ws_client.on(:error) { |err| ws_client_event_handler.on_error(err) }
+      end
```

Comment:
I thought a lot about this code, and breaking out `WebSocketClientEventHandler`.  I think your instinct that this class was doing a bit too much was correct, but there is a better way to slice it up.  

A few things to notice that `WebSocketClientEventHandler` is not the ideal abstraction:

1. It duplicates almost all the arguments of this CommandClass.  
2. The logic that it performs really is the heart of this CommandClass.  It feels like we're separating something that should be a holistic thing.
3. The name is bulky.

As far as trimming the class down, I think the better candidates are:

1. `server_url` -> value object.
2. `timeout` is a purely functional, re-usable logic that doesn't have anything to do with this class per se.  It can be pulled out into something like:
```ruby
      class EnvVarInteger
        def initialize(env:, key:, default: nil)
          @env = env
          @key = key
          @default = default
        end
        
        def value
          return @value if @value
          
          val = @env[key]
          not_provided = val.to_s.strip.empty?
          # If the value is not an integer (eg, "blah"), it will be zero.
          @value = not_provided ? @default : val.to_i
        end
      end
```
and then `timeout` becomes:
```ruby
      def timeout
        @timeout ||= EnvVarInteger.new(
          env: @env,
          key: "KUBE_EXEC_COMMAND_TIMEOUT",
          default: DEFAULT_KUBE_EXEC_COMMAND_TIMEOUT
        ).value
      end
```
and I'm not sure if we need `DEFAULT_KUBE_EXEC_COMMAND_TIMEOUT` to be defined as a class constant versus just being a constant in this method.

Reply:
Oh one more thing I realized:  Rather than make the entire `ENV` struct a dependency, we should pass in just the timeout value and use the default `ENV["KUBE_EXEC_COMMAND_TIMEOUT"]` in the CommandClass initializer.  We can still keep the above logic, we just won't have to deal with the `@env[key]` part anymore.

Reply:
@jonahx  I would add a warning if the value is not an int.

Reply:
@eranha That's a good idea.  Oren and I had previously discussed throwing an error, which was we thought was too heavy-headed in this case.  A warning is a good compromise.